### PR TITLE
feat: add useRef

### PR DESCRIPTION
**Pull request by Caele**
## Motivation

Adds a useRef hook that works similarly to the preact/react counterpart.

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [ ] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [x] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

**Review by VasilisKalaitzis — APPROVED** [review]
Awesome!

File: apis/supernova/src/hooks.js [hunk 1]
```diff
@@ -398,6 +398,31 @@ export function useMemo(fn, deps) {
...
+ * const timesRendered = useRef(0);
+ *
+ * useEffect(() => {
+ *  render(layout);
+ *  // increments the render counter, a useState would trigger another render
+ *  timesRendered.current += 1;
+ * },[layout]);
+ *
+ */
+export function useRef(initialValue) {
```
Comment (T-Wizard) [inline apis/supernova/src/hooks.js:422]: do add jsdoc to give it a proper generic type

File: apis/supernova/src/__tests__/hooks.test.js [hunk 2]
```diff
@@ -300,6 +301,59 @@ describe('hooks', () => {
...
+
+      run(c);
+      expect(countValue.current).toBe(7);
+      countValue.current = 11;
+      expect(num).toBe(1);
+      await frame();
+      expect(num).toBe(1);
+      expect(countValue.current).toBe(11);
+    });
+  });
```
Comment (johanlahti) [inline apis/supernova/src/__tests__/hooks.test.js:355]: Is it possible to write a test which verifies that an updated does not trigger a re-render? (maybe not necessary…)

**Review by johanlahti — APPROVED** [review]

File: apis/supernova/src/__tests__/hooks.test.js (hunk 2 above)
Comment (Caele) [reply apis/supernova/src/__tests__/hooks.test.js:355]: That last test should cover that, even if it is a bit hard/strange to test it properly. You'd basically need to test a use case where the developer used it incorrectly.

**Review by quanho — APPROVED** [review]

File: apis/stardust/types/index.d.ts [hunk 3]
```diff
@@ -41,6 +41,17 @@ export function useEffect(effect: stardust.EffectCallback, deps?: any[]): void;
...
+/**
+ * Creates a reference to a value not needed for rendering
+ * 
+ * While Nebula does not have a virtual DOM, it is still useful
+ * to have a reference to an object that is retained across
+ * renders and in it self does not trigger a render.
+ * @param initialValue The initial value.
+ */
+export function useRef<R>(initialValue: R): {
+};
```
Comment (T-Wizard) [inline apis/stardust/types/index.d.ts:53]: return type is still not correct
(maybe could be fixed by adding and using a generic Ref type, similar to SetStateFn used by useState?)

Comment (Caele) [reply apis/stardust/types/index.d.ts]: tricky to get right, used an interface now

**Review by T-Wizard — APPROVED** [review]

**Pull request merged**